packages/storage/src/chrome/base.ts

```diff
@@ -1,9 +1,15 @@
 import { EmptyObject, isEmptyObj } from '@penumbra-zone/types/src/utility';
 
+type Listener = (changes: Record<string, { oldValue?: unknown; newValue?: unknown }>) => void;
+
 export interface IStorage {
   get(key: string): Promise<Record<string, unknown>>;
   set(items: Record<string, unknown>): Promise<void>;
   remove(key: string): Promise<void>;
+  onChanged: {
+    addListener(listener: Listener): void;
+    removeListener(listener: Listener): void;
+  };
 }
```

Comment:
I feel like there is a missed opportunity to use a generic here. For instance, `ExtensionStorage` can pass `T` to it right? And then we can use that to improve these methods. Or maybe not, up to you!

Reply:
Just tried this, and it looks like that's not possible because `chrome.storage.local` doesn't support generics :( so we can't actually get type-safety here — we have to treat the changed keys as just `string` rather than `keyof T`.